cuda/base/device_guard.hpp

```diff
@@ -57,10 +57,13 @@ namespace cuda {
  */
 class device_guard {
 public:
-    device_guard(int device_id)
+    device_guard(int device_id) : original_device_id{}, need_reset{}
```

Comment:
For readability, might be better to be explicit and initialize it to `false` ? 
```suggestion
    device_guard(int device_id) : original_device_id{}, need_reset{false}
```